### ISSUE TITLE
Setting Authorization headers in tests

### ISSUE THREAD

**Issue by mattclark2500** [chaijs/chai-http#159]
I have a test that looks like this:

```
const jwt = 'bearer eg.someLongStringThatIsActuallyAValidJWT';

it('it should GET all the users', done => {
      chai.request(server)
        .get('/v1/user')
        .set('Authorization', jwt)
        .end((err, res) => {
          res.should.have.status(200);
          done();
        });
    });

```
This route I'm testing requires a JWT to be passed in the request header. I'm using the [koa-jwt](https://www.npmjs.com/package/koa-jwt) module to handle this in my app. Testing this route with a REST client, and passing in the appropriate headers returns the expected result with a 200 status.

When I run my tests, however, all the tests on the protected routes fail because they return with a status code of 401 (not authorized). I have tried using the .set() method to pass in the same headers (same key & same JWT). But the tests seem to ignore the headers. Any suggestions as to where I might be going wrong?

**Comment by vieiralucas** [comment]
Hello @mattclark2500, thank you for your issue

I tried to reproduced your problem with the following code

Using `chai v3.5.0` and `chai-http v3.0.0`
```js
const express = require('express')

const app = express()

app.get('/', (req, res) => {
  if (req.headers.authorization === 'super secure auth') {
    res.status(200).json({ msg: 'welcome' })
    return
  }

  res.status(401).json({ msg: 'get out' })
})

app.listen(8080, () => {
  console.log('listening')
})

const chai = require('chai')
const chaiHttp = require('chai-http')

const expect = chai.expect

chai.use(chaiHttp)

describe('GET /', () => {
  it('send 200 if authorization is provided', () => {
    return chai.request(app)
      .get('/')
      .set('Authorization', 'super secure auth')
      .then(res => {
        expect(res).to.have.status(200)
      });
  })
})
```

If I run this code with mocha, It says that my tests pass.

It might be something wrong with your server when retrieving the `jsonwebtoken` from the `Authorization` header.
Can you provide a example public repo that reproduces the problem?

Thank you!

**Comment by vieiralucas** [comment]
@mattclark2500 I also created an example on RunKit using `jsonwebtoken` but with `express`, since I'm not familiar with `koa`
You can visit it here: https://runkit.com/vieiralucas/chai-http-159

**Comment by mattclark2500** [comment]
@vieiralucas Many thanks for your support on this. Am afraid it was a bad case of user error, which should have been obvious to me much earlier. The JWT secret was set as an environment variable for my dev config, but wasn't set for my test config. I eventually found it when I delved into the koa-jwt module and found the error thrown was actually that the secret was missing. Now works wonderfully.. once again, many thanks for your help!
